tiledb/ml/models/base.py

```diff
@@ -9,13 +15,125 @@ class TileDBModel(abc.ABC):
+
+    def get_s3_prefix(self):
+        """
+        Get S3 path from the user profile or organization profile
+        :return: s3 path or error
+        """
+        try:
+            profile = tiledb.cloud.client.user_profile()
+
+            if self.namespace == profile.username:
+                if profile.default_s3_path is not None:
+                    return os.path.join(profile.default_s3_path, CLOUD_MODELS)
+            else:
+                organization = tiledb.cloud.client.organization(self.namespace)
+                if organization.default_s3_path is not None:
+                    return os.path.join(organization.default_s3_path, CLOUD_MODELS)
+        except tiledb.cloud.tiledb_cloud_error.TileDBCloudError as e:
+            raise HTTPError(
+                400,
+                "Error fetching user default s3 path for new ML models {}".format(str(e)),
+            )
+
+        return None
+
+    def get_s3_credentials(self):
+        """
+        Get credentials for default S3 path from the user profile or organization profile
+        :return: s3 credentials or error
+        """
+        try:
+            profile = tiledb.cloud.client.user_profile()
+
+            if self.namespace == profile.username:
+                if profile.default_s3_path_credentials_name is not None:
+                    return profile.default_s3_path_credentials_name
+            else:
+                organization = tiledb.cloud.client.organization(self.namespace)
+                if organization.default_s3_path_credentials_name is not None:
+                    return organization.default_s3_path_credentials_name
+        except tiledb.cloud.tiledb_cloud_error.TileDBCloudError as e:
+            raise HTTPError(
+                400,
+                "Error fetching default credentials for {} default s3 path for new ML models {}".format(
+                    self.namespace, str(e)
+                ),
+            )
+
+        return None
+
+    def _model_exists(self):
+        """
+        Check if an model exists in TileDB Cloud
+        :return:
+        """
+        try:
+            tiledb.cloud.array.info(self.tiledb_uri)
+            return True
+        except tiledb.cloud.tiledb_cloud_error.TileDBCloudError as e:
+            if str(e) == "Array or Namespace Not found":
+                return False
+
+        return False
+
+    def get_model_info(self):
+        """
+        Check if an model exists in TileDB Cloud
+        :return:
+        """
+        try:
+            model_info = tiledb.cloud.array.info(self.tiledb_uri)
+            return model_info
+        except tiledb.cloud.tiledb_cloud_error.TileDBCloudError as e:
+            if str(e) == "Array or Namespace Not found":
+                return None
+
+        return None
+
+    def delete_model(self):
+        """Delete the file or directory at path."""
+        try:
+            return tiledb.cloud.array.delete_array(self.tiledb_uri, "application/x-ipynb+json")
```

Comment:
Shouldn't this work for local (non tiledb-cloud) arrays?

Reply:
Added local case